File: NapakalakiRubyJavi/lib/monster.rb
#Clase Monster
#Autor: Javier Aranda Izquierdo
require relative "prize"
require relative "bad_consequence"

class Monster
  def initialize(name,combarLevel,prize,badConsequence)
    @name=name
    @combarLevel=combarLevel
    @prize=prize
    @badConsequence=badConsequence
    
  end
  
# Consulores basicos
attr_accesor:name
attr_accesor:combatLEvel
attr_accesor:prize
attr_accesor:badConsequenze

end

def to_s
  "Nombre del monstruo: #{name} \nNivel de combate: #{level}"
end

File: Napakalaki-Java/src/napakalaki/Monster.java
package napakalaki;

/**
 *
 * @author Javier Aranda
 */
public class Monster {
    
    //Attrb
    private String name;
    private int combatLevel;
    private BadConsequence badConsequence;
    private Prize prize;
    
    //Constrc
    public Monster(String name, int level, BadConsequence bc, Prize prize){
        this.name=name;
        combatLevel=level;
        badConsequence=bc;
        this.prize=prize;
    }
    
    public String getName(){
        return name;
    }
    
    public int getCombatLevel(){
        return combatLevel;
    }
    
    public BadConsequence getBadConsequence(){
        return badConsequence;
    }
    
    public Prize getPrize(){
        return prize;
    }
    
    public String toString(){
        return "Nombre = " + (name) + ", Nivel de combate = " + Integer.toString(combatLevel);
    }
    
}
